ruby_on_rails/advanced_forms_and_activerecord/installing_psql.md

```diff
@@ -0,0 +1,239 @@
+
+### Environment Variables
+The foundation of any environment is the operating system which provides ways to store local variables. Not surprisingly, these are known as environment variables.
+
+Why are environment variables important? After all, the information could be added to the git repository (and by extension Github). While tools like git and Github are excellent for storing and sharing code, sensitive information like API keys or passwords should be kept private. Using environment variables, this type of private data can only be accessed locally which means the application gets the information it needs without exposing it to the world.
+
+Rails uses the environment to set up its own internal environment variable. The framework will look for a `RAILS_ENV` variable in the environment and set its own env that way. If it can't find the variable, then [Rails will assume it is in a development environment](https://github.com/rails/rails/blob/main/railties/lib/rails.rb#L67). Isn't that cool?
+
+We will use an environment variable to set up PostgreSQL credentials for our Rails apps.
+
+### Installing PostgreSQL
+
+If you've been following the curriculum, then you should already have all the supporting bits and pieces installed. If not, please go back and refer to those lessons before continuing. You'll want to confirm that node.js, Ruby, Rails and yarn are all installed.
+
+Pick your operating system below to get the relevant steps for installing PostgreSQL.
+
+<details markdown="block">
+<summary class="dropDown-header">Ubuntu
+</summary>
+
+### Step 1: Make sure the system is up to date
+Before installing PostgreSQL, it's a good idea to make sure the operating system is up to date.
+
+In Ubuntu, press `Ctrl + Alt + T` to open a new terminal. As a quick reminder, copy and pasting to the terminal in Ubuntu requires doing `Ctrl + Shift + C` and `Ctrl + Shift + V`.
+
+<div class="lesson-note" markdown="1">
+#### Note
+We will be using `sudo` quite a bit. As you follow these instructions, you may be (re-)prompted for your password. Remember that there will be no visual feedback when you type in your password.
+</div>
+
+To update our system, type in these commands:
+
+~~~~bash
+sudo apt update
+sudo apt upgrade
+~~~~
+
+Be sure to review any prompts before pressing `Y` and `Enter`. While they typically advise you about additional space required for updates, this might not be the case!
+
+### Step 2: Install the PostgreSQL packages
+After our system is up to date, we will install the packages for PostgreSQL.
+
+~~~bash
+sudo apt install postgresql postgresql-contrib libpq-dev
+~~~
+
+After installation is complete, let's start the server using this command:
+
+~~~bash
+sudo systemctl start postgresql.service
+~~~
+
+If you are unsure about whether postgre is active, it's possible to check with this command:
+
+~~~bash
+service postgresql status
+~~~
+
+Got an error, or don't see an active service? Come visit the Discord for some help!
```

Comment:
Throughout the curriculum I think Discord is usually a link, so it might make sense to keep doing that. There's two other mentions of Discord in this lesson that need a link, I won't be creating separate comments for them.
```suggestion
Got an error, or don't see an active service? Come visit the [Discord](https://discord.gg/V75WSQG) for some help!
```